### PR TITLE
Improve grib file processing speed

### PR DESCRIPTION
**Pull request by pramodg**
Copy the grib data variables to in memory to avoid disk access on subsequent reads.

### CODE REVIEW

File: weather_mv/loader_pipeline/pipeline.py [hunk 1]
```diff
@@ -57,6 +57,16 @@ def _prod(xs: t.Iterable[int]) -> int:
     return reduce(operator.mul, xs, 1)
 
 
+def _make_grib_dataset_inmem(grib_ds: xr.Dataset) -> xr.Dataset:
```
Comment (alxmrs) [inline weather_mv/loader_pipeline/pipeline.py:60]: I don't think this function is necessary. xarray's `open_dataset` provides the `cache` argument, which can be used to open data in memory: https://xarray.pydata.org/en/stable/generated/xarray.open_dataset.html

File: weather_mv/loader_pipeline/pipeline_test.py [hunk 2]
```diff
@@ -301,6 +315,24 @@ def test_multiple_editions(self):
         }
         self.assertRowsEqual(actual, expected)
 
+    @_handle_missing_grib_be
+    def test_timing_profile(self):
```
Comment (alxmrs) [inline weather_mv/loader_pipeline/pipeline_test.py:319]: I like the profiling test!

**Comment by pramodg** [comment]
As a simple comparison on a local machine, for a single variable read for 1K coordinates the speed up is ~40 times. (Not rigorously tested, but rough ballpark).

File: weather_mv/loader_pipeline/pipeline.py (hunk 1 above)
Comment (alxmrs) [reply weather_mv/loader_pipeline/pipeline.py:60]: The `chunks` argument may also help here.

Comment (pramodg) [reply weather_mv/loader_pipeline/pipeline.py:60]: cache=True is lazy caching, while it does improve read times, its much smaller improvement (only ~2x) on the data i tested. 
Unless we have a memory concern, don't think using that path instead is much more helpful.

Comment (alxmrs) [reply weather_mv/loader_pipeline/pipeline.py:60]: I see. I'm glad that you tested this pathway. In that case, I'm happy with the solution you provided. I'll take one more look at the CL to see if I can find any issues.

File: weather_mv/loader_pipeline/pipeline.py [hunk 3]
```diff
@@ -66,7 +76,8 @@ def __open_dataset_file(filename: str) -> xr.Dataset:
             raise
     # Trying with explicit engine for cfgrib.
     try:
-        return xr.open_dataset(filename, engine='cfgrib', backend_kwargs={'indexpath': ''})
+        return _make_grib_dataset_inmem(
```
Comment (alxmrs) [inline weather_mv/loader_pipeline/pipeline.py:79]: Can you add a flag so the user can control if the dataset should be opened in memory? I can see this tool being used on O(100GB) files, where users would like to turn this off.

File: weather_mv/loader_pipeline/pipeline_test.py [hunk 4]
```diff
@@ -301,6 +315,23 @@ def test_multiple_editions(self):
...
+        self.test_data_path = f'{self.test_data_folder}/test_data_grib_single_timestep'
+        counter = 0
+        i = extract_rows(self.test_data_path)
+        # Read once to avoid counting dataset open times, etc.
+        _ = next(i)
+        with timing('Loop'):
+            for v in i:
+                # Don't do everything, 10K coordinates is enough for testing.
+                if counter > 10000:
```
Comment (alxmrs) [inline weather_mv/loader_pipeline/pipeline_test.py:328]: nit: `>=`, since it's zero indexed.

File: weather_mv/loader_pipeline/pipeline.py (hunk 3 above)
Comment (alxmrs) [reply weather_mv/loader_pipeline/pipeline.py:79]: I'm open to creating an issue to address this in the future.

**Review by alxmrs — APPROVED** [review]
Added a recommendation for usability. Otherwise, LGTM.

File: weather_mv/loader_pipeline/pipeline.py (hunk 3 above)
Comment (pramodg) [reply weather_mv/loader_pipeline/pipeline.py:79]: Agreed, let's open an issue and get this submitted for now.

File: weather_mv/loader_pipeline/pipeline_test.py (hunk 4 above)
Comment (pramodg) [reply weather_mv/loader_pipeline/pipeline_test.py]: done

**Pull request merged**